src/base/SelfPermit.sol

```diff
@@ -1,52 +0,0 @@
-// SPDX-License-Identifier: UNLICENSED
-pragma solidity ^0.8.19;
-
-import {IERC20} from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
-import {IERC20Permit} from "@openzeppelin/contracts/token/ERC20/extensions/IERC20Permit.sol";
-
-import {IERC20PermitAllowed} from "../interfaces/external/IERC20PermitAllowed.sol";
-import {ISelfPermit} from "../interfaces/ISelfPermit.sol";
-
-/// @title Self Permit
-/// @notice Functionality to call permit on any EIP-2612-compliant token for use in the route
-/// @dev These functions are expected to be embedded in multicalls to allow EOAs to approve a contract and call a function
-/// that requires an approval in a single transaction.
-abstract contract SelfPermit is ISelfPermit {
```

Comment:
what did this used to be used for?

Reply:
V3PositionManager used it.. I think to permit the manager. We will have something that works with permit2 to permit this contract as a spender on permit2